src/js/components/DataTable/Header.js

```diff
@@ -261,17 +254,22 @@ const Header = forwardRef(
                       ? 'unselect all'
                       : 'select all'
                   }
+                  // should the logic for these states live here or
```

Comment:
I'm supportive of pulling them above so the return is cleaner, but also fine leaving as is if you don't want to add more changes to the PR.

Once you decide, we can remove this comment.

Reply:
done